### PR TITLE
api: v1alpha2 version added. part 1

### PR DESCRIPTION
**Pull request by alebedev87**
Follow-up of https://github.com/openshift/node-observability-operator/pull/83. `NodeObservability.Spec.NodeSelector` field replaced the existing `NodeObservability.Spec.Labels` which broke the compatibility. This PR aims as adding a new API version `v1alpha2` which re-introduces the breaking API feature. [Kubebuilder guide](https://book.kubebuilder.io/multiversion-tutorial/tutorial.html) was followed.

**Things changed**

- `v1alpha2` version added by simply coping from `v1alpha1` at its state as per https://github.com/openshift/node-observability-operator/pull/83
- `v1alpha1` reverted to the state before https://github.com/openshift/node-observability-operator/pull/83
- Manifests updated (`v1alpha2` is the storage version for all the types)
- `main.go` is now in the root directory as per the new kubebuilder layout
- Controllers code migrated to `v1apha2` version everywhere
- Conversion logic added but not enabled in the manifests
- `AllNamespaces` install mode added to CSV. It will be necessary for the conversion webhook to work.

**Note**
- Conversion webhook cannot be added at this PR as it requires `AllNamespaces` installmode only. The Prow config is not yet ready for it, so this PR introduces both modes `OwnNamespace` and `AllNamespaces`. The `OwnNamespace` mode will be removed along with the addition of the conversion webhook in [the part2 PR](https://github.com/openshift/node-observability-operator/pull/91).
- e2e test and the convenience script for CA injection are coming in [the part2 PR](https://github.com/openshift/node-observability-operator/pull/91) too.

### CODE REVIEW

**Comment by ayesha54** [comment]
great effort @alebedev87

File: main.go [hunk 1]
```diff
@@ -64,20 +65,25 @@ var (
 func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 	utilruntime.Must(nodeobservabilityv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(nodeobservabilityv1alpha2.AddToScheme(scheme))
 	utilruntime.Must(securityv1.AddToScheme(scheme))
 	utilruntime.Must(rbacv1.AddToScheme(scheme))
 	utilruntime.Must(mcv1.AddToScheme(scheme))
+	utilruntime.Must(nodeobservabilityv1alpha2.AddToScheme(scheme))
```
Comment (thejasn) [inline main.go:72]: Duplicate.

File: main.go [hunk 2]
```diff
@@ -86,6 +92,7 @@ func main() {
 	flag.StringVar(&tokenFile, "token-file", defaultTokenFile, "The path of the service account token.")
 	flag.StringVar(&caCertFile, "ca-cert-file", defaultCACertFile, "The path of the CA cert of the Agents' signing key pair.")
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false, "Enable leader election for controller manager. "+"Enabling this will ensure there is only one active controller manager.")
+	flag.BoolVar(&enableWebhook, "enable-webhook", false, "Enable the webhook server(s). Defaults to true.")
```
Comment (thejasn) [inline main.go:95]:
```suggestion
	flag.BoolVar(&enableWebhook, "enable-webhook", true, "Enable the webhook server(s). Defaults to true.")
```

**Review by thejasn — CHANGES_REQUESTED** [review]
Just a couple of things.

File: main.go (hunk 2 above)
Comment (alebedev87) [reply main.go:95]: This was done on purpose, I didn't want to start webhook server until the conversion config is not there yet. Will do it [in the followup PR](https://github.com/openshift/node-observability-operator/pull/91).

Comment (thejasn) [reply main.go]: ahh, that makes sense.

**Pull request merged**